### PR TITLE
Add tests for thread interrupt in finally blocks

### PR DESCRIPTION
**Pull request by kouvel**
Related to dotnet/coreclr#8953

### CODE REVIEW

**Comment by kouvel** [comment]
Needs CoreCLR PR to be merged and packages updated before this can be merged.

@jkotas @gkhanna79

File: src/System.Threading.Thread/tests/ThreadTests.netstandard1.7.cs [hunk 1]
```diff
@@ -664,6 +666,32 @@ public static void InterruptTest()
...
         [Fact]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework)]
+        public static void InterruptInFinallyBlockTest_SkipOnDesktopFramework()
+        {
+            // A wait in a finally block can be interrupted. The desktop framework applies the same rules as thread abort, and
+            // does not allow thread interrupt in a finally block. There is nothing special about thread interrupt that requires
+            // not allowing it in finally blocks, so this behavior has changed in .NET Core.
+            var continueThread = new AutoResetEvent(false);
+            Action waitForThread;
+            var t =
```
Comment (stephentoub) [inline src/System.Threading.Thread/tests/ThreadTests.netstandard1.7.cs]: Nit: var => Thread

Comment (kouvel) [inline src/System.Threading.Thread/tests/ThreadTests.netstandard1.7.cs]: Done

**Comment by jkotas** [comment]
@kouvel Could you please resolve the merge conflict? I have just merged CoreCLR update so this should pass now.

**Comment by kouvel** [comment]
It looks like it hasn't picked up the updated CoreCLR yet, the test is passing against latest CoreCLR locally but failing with the CoreCLR package CoreFX is currently using

**Comment by kouvel** [comment]
@dotnet-bot test this please

**Comment by JonHanna** [comment]
Is there a good alternative method to guard an section against a thread abort?

**Comment by kouvel** [comment]
Thread abort is not supported by .NET Core

**Comment by jkotas** [comment]
> Is there a good alternative method to guard an section against a thread abort?

Wrapping the code by finally block still works fine for runtimes that support thread abort. It is no-op for .NET Core.

**Comment by stephentoub** [comment]
@dotnet-bot test this please (now that #15219 has been fixed)

**Pull request merged**